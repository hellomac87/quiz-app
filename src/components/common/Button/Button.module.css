.container {
    @apply inline-flex items-center justify-center bg-primary500 h-12 px-6 text-white rounded-md text-base;
    @apply transition-all;
    @apply hover:bg-primary700;
}

.fullWidth {
    @apply w-full;
}
