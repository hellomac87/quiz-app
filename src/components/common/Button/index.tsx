import React, { ButtonHTMLAttributes } from 'react';
import clsx from 'clsx';

import styles from './Button.module.css';

type Props = ButtonHTMLAttributes<HTMLButtonElement> & {
    width?: number;
    fullWidth?: boolean;
    variant?: 'fill' | 'outline';
};

function Button({
    children,
    width,
    fullWidth = false,
    variant = 'fill',
    ...rest
}: React.PropsWithChildren<Props>): JSX.Element {
    return (
        <button
            {...rest}
            className={clsx(styles.container, styles[variant], { [styles.fullWidth]: fullWidth })}
            style={{ width }}
        >
            {children}
        </button>
    );
}

export default Button;
