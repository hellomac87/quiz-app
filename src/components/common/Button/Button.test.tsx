import renderer from 'react-test-renderer';

import Button from 'src/components/common/Button';

const onClick = () => {
    console.log('onClick');
};

describe('Button test', () => {
    test('snapshot: no Props', () => {
        const tree = renderer.create(<Button>Button</Button>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('snapshot: onClick', () => {
        const tree = renderer.create(<Button onClick={onClick}>Button</Button>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
