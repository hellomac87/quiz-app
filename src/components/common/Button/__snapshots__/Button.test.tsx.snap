// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test snapshot: no Props 1`] = `
<button
  className="container fill"
  style={
    Object {
      "width": undefined,
    }
  }
>
  Button
</button>
`;

exports[`Button test snapshot: onClick 1`] = `
<button
  className="container fill"
  onClick={[Function]}
  style={
    Object {
      "width": undefined,
    }
  }
>
  Button
</button>
`;
